CMakeLists.txt

```diff
@@ -307,7 +307,15 @@ if (BENCHMARK_ENABLE_TESTING)
   if (BENCHMARK_ENABLE_GTEST_TESTS AND
       NOT (TARGET gtest AND TARGET gtest_main AND
            TARGET gmock AND TARGET gmock_main))
-    include(GoogleTest)
+    find_package(GTest QUIET)
```

Comment:
https://github.com/google/googletest/blob/master/googletest/README.md#incorporating-into-an-existing-cmake-project suggests at least `find_package(GTest CONFIG REQUIRED)`

Reply:
`REQUIRED` will require this library and users will need to install gtest-devel (gtest-dev) packages. We don't need such behavior.

`QUIET` means that it will try to find the packaged version and, if not found, fallback to `include(GoogleTest)` - use download/bundled.

Reply:
We don't use `CONFIG` option, because we want to use build-in CMake's implementation of FindGTest.cmake. It supports exporting both GTest and GMock. Both are needed and used in tests.

Reply:
It works fine. Tested on Fedora 35:

```text
+ /usr/bin/ctest --output-on-failure --force-new-ctest-process -j4
Test project /builddir/build/BUILD/benchmark-1.6.0/x86_64-redhat-linux-gnu
      Start  1: benchmark
      Start  2: filter_simple
      Start  3: filter_simple_list_only
      Start  4: filter_simple_negative
 1/68 Test  #3: filter_simple_list_only ....................   Passed    0.00 sec
      Start  5: filter_simple_negative_list_only
 2/68 Test  #4: filter_simple_negative .....................   Passed    0.00 sec
      Start  6: filter_suffix
 3/68 Test  #5: filter_simple_negative_list_only ...........   Passed    0.00 sec
      Start  7: filter_suffix_list_only
 4/68 Test  #6: filter_suffix ..............................   Passed    0.00 sec
      Start  8: filter_suffix_negative
 5/68 Test  #7: filter_suffix_list_only ....................   Passed    0.00 sec
      Start  9: filter_suffix_negative_list_only
 6/68 Test  #8: filter_suffix_negative .....................   Passed    0.00 sec
      Start 10: filter_regex_all
 7/68 Test  #9: filter_suffix_negative_list_only ...........   Passed    0.00 sec
      Start 11: filter_regex_all_list_only
 8/68 Test #10: filter_regex_all ...........................   Passed    0.00 sec
      Start 12: filter_regex_all_negative
 9/68 Test #11: filter_regex_all_list_only .................   Passed    0.00 sec
      Start 13: filter_regex_all_negative_list_only
10/68 Test #12: filter_regex_all_negative ..................   Passed    0.00 sec
      Start 14: filter_regex_blank
11/68 Test #13: filter_regex_all_negative_list_only ........   Passed    0.00 sec
      Start 15: filter_regex_blank_list_only
12/68 Test #14: filter_regex_blank .........................   Passed    0.00 sec
      Start 16: filter_regex_blank_negative
13/68 Test  #2: filter_simple ..............................   Passed    0.01 sec
      Start 17: filter_regex_blank_negative_list_only
14/68 Test #15: filter_regex_blank_list_only ...............   Passed    0.00 sec
      Start 18: filter_regex_none
15/68 Test #16: filter_regex_blank_negative ................   Passed    0.00 sec
      Start 19: filter_regex_none_list_only
16/68 Test #17: filter_regex_blank_negative_list_only ......   Passed    0.00 sec
      Start 20: filter_regex_none_negative
17/68 Test #18: filter_regex_none ..........................   Passed    0.00 sec
      Start 21: filter_regex_none_negative_list_only
18/68 Test #19: filter_regex_none_list_only ................   Passed    0.00 sec
      Start 22: filter_regex_wildcard
19/68 Test #20: filter_regex_none_negative .................   Passed    0.00 sec
      Start 23: filter_regex_wildcard_list_only
20/68 Test #21: filter_regex_none_negative_list_only .......   Passed    0.00 sec
      Start 24: filter_regex_wildcard_negative
21/68 Test #22: filter_regex_wildcard ......................   Passed    0.00 sec
      Start 25: filter_regex_wildcard_negative_list_only
22/68 Test #23: filter_regex_wildcard_list_only ............   Passed    0.00 sec
      Start 26: filter_regex_begin
23/68 Test #24: filter_regex_wildcard_negative .............   Passed    0.00 sec
      Start 27: filter_regex_begin_list_only
24/68 Test #25: filter_regex_wildcard_negative_list_only ...   Passed    0.00 sec
      Start 28: filter_regex_begin_negative
25/68 Test #26: filter_regex_begin .........................   Passed    0.00 sec
      Start 29: filter_regex_begin_negative_list_only
26/68 Test #27: filter_regex_begin_list_only ...............   Passed    0.00 sec
      Start 30: filter_regex_begin2
27/68 Test #28: filter_regex_begin_negative ................   Passed    0.00 sec
      Start 31: filter_regex_begin2_list_only
28/68 Test #29: filter_regex_begin_negative_list_only ......   Passed    0.00 sec
      Start 32: filter_regex_begin2_negative
29/68 Test #30: filter_regex_begin2 ........................   Passed    0.00 sec
      Start 33: filter_regex_begin2_negative_list_only
30/68 Test #31: filter_regex_begin2_list_only ..............   Passed    0.00 sec
      Start 34: filter_regex_end
31/68 Test #32: filter_regex_begin2_negative ...............   Passed    0.00 sec
      Start 35: filter_regex_end_list_only
32/68 Test #33: filter_regex_begin2_negative_list_only .....   Passed    0.00 sec
      Start 36: filter_regex_end_negative
33/68 Test #34: filter_regex_end ...........................   Passed    0.00 sec
      Start 37: filter_regex_end_negative_list_only
34/68 Test #35: filter_regex_end_list_only .................   Passed    0.00 sec
      Start 38: options_benchmarks
35/68 Test #36: filter_regex_end_negative ..................   Passed    0.00 sec
      Start 39: basic_benchmark
36/68 Test #37: filter_regex_end_negative_list_only ........   Passed    0.00 sec
      Start 40: repetitions_benchmark
37/68 Test #40: repetitions_benchmark ......................   Passed    0.02 sec
      Start 41: diagnostics_test
38/68 Test #41: diagnostics_test ...........................   Passed    0.04 sec
      Start 42: skip_with_error_test
39/68 Test #42: skip_with_error_test .......................   Passed    0.23 sec
      Start 43: donotoptimize_test
40/68 Test #43: donotoptimize_test .........................   Passed    0.00 sec
      Start 44: fixture_test
41/68 Test #44: fixture_test ...............................   Passed    0.03 sec
      Start 45: register_benchmark_test
42/68 Test #45: register_benchmark_test ....................   Passed    0.00 sec
      Start 46: map_test
43/68 Test #46: map_test ...................................   Passed    0.25 sec
      Start 47: multiple_ranges_test
44/68 Test #39: basic_benchmark ............................   Passed    0.75 sec
      Start 48: args_product_test
45/68 Test #47: multiple_ranges_test .......................   Passed    0.34 sec
      Start 49: link_main_test
46/68 Test #49: link_main_test .............................   Passed    0.02 sec
      Start 50: reporter_output_test
47/68 Test #48: args_product_test ..........................   Passed    0.26 sec
      Start 51: templated_fixture_test
48/68 Test #51: templated_fixture_test .....................   Passed    0.03 sec
      Start 52: user_counters_test
49/68 Test #50: reporter_output_test .......................   Passed    0.25 sec
      Start 53: perf_counters_test
50/68 Test #53: perf_counters_test .........................   Passed    0.00 sec
      Start 54: internal_threading_test
51/68 Test #52: user_counters_test .........................   Passed    0.23 sec
      Start 55: report_aggregates_only_test
52/68 Test #55: report_aggregates_only_test ................   Passed    0.00 sec
      Start 56: display_aggregates_only_test
53/68 Test #56: display_aggregates_only_test ...............   Passed    0.00 sec
      Start 57: user_counters_tabular_test
54/68 Test #57: user_counters_tabular_test .................   Passed    0.23 sec
      Start 58: user_counters_thousands_test
55/68 Test #58: user_counters_thousands_test ...............   Passed    0.01 sec
      Start 59: memory_manager_test
56/68 Test #59: memory_manager_test ........................   Passed    0.03 sec
      Start 60: cxx03
57/68 Test #60: cxx03 ......................................   Passed    0.23 sec
      Start 61: complexity_benchmark
58/68 Test #38: options_benchmarks .........................   Passed    2.22 sec
      Start 62: benchmark_gtest
59/68 Test #62: benchmark_gtest ............................   Passed    0.00 sec
      Start 63: benchmark_name_gtest
60/68 Test #63: benchmark_name_gtest .......................   Passed    0.00 sec
      Start 64: benchmark_random_interleaving_gtest
61/68 Test #64: benchmark_random_interleaving_gtest ........   Passed    0.00 sec
      Start 65: commandlineflags_gtest
62/68 Test #65: commandlineflags_gtest .....................   Passed    0.00 sec
      Start 66: statistics_gtest
63/68 Test #66: statistics_gtest ...........................   Passed    0.00 sec
      Start 67: string_util_gtest
64/68 Test #67: string_util_gtest ..........................   Passed    0.00 sec
      Start 68: perf_counters_gtest
65/68 Test #68: perf_counters_gtest ........................   Passed    0.00 sec
66/68 Test #61: complexity_benchmark .......................   Passed    1.30 sec
67/68 Test #54: internal_threading_test ....................   Passed    1.94 sec
68/68 Test  #1: benchmark ..................................   Passed    3.19 sec

100% tests passed, 0 tests failed out of 68

Total Test time (real) =   3.19 sec
```